core/query/sql.ml

```diff
@@ -26,9 +26,16 @@ and select_clause = (base * string) list * from_clause list * base * base list
 and from_clause =
   | TableRef of table_name * Var.var
   | Subquery of query * Var.var
+and sql_like =
```

Comment:
I'm curious what is the benefit of having a separate case for LIKE strings as opposed to just using the string constant, ordinary projection, and string concatenation operators.  Clearly doing it this way makes it easier to recognize where match expressions have been translated to (LIKE) strings since regexp compilation is the only place where we do that.  However, what would break (or be unpleasantly harder) if we just compiled regexps to the existing Constant(String -), Project and Append operations?